### ISSUE TITLE
[Question] Multiple domain names

### ISSUE THREAD

**Issue by ddk9499** [docker-mailserver/docker-mailserver#2344]
# Subject
Other

## Description
Can we use multiple domains in our emails?
1. `user@domain.com`
2. `another_user@another-domain.com`

If it's possible, how to achieve this? I can not find about this in documentation

**Comment by georglauterbach** [comment]
You are wasting maintainers' time:

https://github.com/docker-mailserver/docker-mailserver/blob/701037d3b2909db5b268ae4dbc171d70921600c3/README.md?plain=1#L181-L189

**Comment by Omemanti** [comment]
I understand that it is very much documented.
but I'm conflicted, so Id like to ask for some clarification:

Under **usage**:

>     Edit docker-compose.yml to your liking
>     substitute mail (hostname) and example.com (domainname) according to your FQDN

Which domainname should I use when I have 2 completely different domains with separate mail boxes.

thnx

**Comment by ontheair81** [comment]
> Which domainname should I use when I have 2 completely different domains with separate mail boxes.

The domain name you choose there, will be the domain name your clients (Thunderbird, for example) connect to (IMAP and SMTP settings). It does not need to be the same as the domain name for your mailboxes.

For example:
I have several different domains, used for e-mail, with many users. For convenience I decided to get another additional generic domain. This generic domain I am using in `docker-compose.yml`, and this domain users will use when configuring their client software.

Of course, that is just my way of doing it, but it maybe gives you a hint for your situation.

**Comment by Omemanti** [comment]
Makes sence, probably not the "cleanest"  way, but this will work for me :) 

Thanks!

**Comment by ontheair81** [comment]
In fact, this is how the big mail providers are doing it, too.

Let's say you are ordering e-mail services at a big provider (e.g. Host Europe) with your personal domain.
The provider will set up a mailserver for you and your domain, but the login to this mailserver is provided by a generic domain (e.g. mailserver12345.hosteurope.com).

So this is basically how I do it and it will work for you, too. Of course you are free to make it your own way, but I would not call it an "unclean" way ;-)

**Comment by Omemanti** [comment]
yes, maybe "unclean" is not the right choose of words, even if using mailserver12345 is best practice, it looks a bit silly don't you think ;)  

but again, this will work.  thanks! 👍

**Comment by rohankm** [comment]
Hi im using docker compose and how to add two domain names in 
`version: '3.8'

services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver
    hostname: mail
    # Change this to your domain, it is used for your email accounts (eg: user@example.com):
    domainname: example.com`

**Comment by ciekawy** [comment]
in particular I am trying to figure out to use single DMS to send emails for my two domains and currently after adding second domain in general it works but for some reason emails are going to spam folder in gmail. WIth various headers check the only actual warning I see is that the sender domains does not match for `smtp.mailfrom`  and 'from' headers which is true as 
`smtp.mailfrom` is sth like 'smtp.mailfrom="SRS0=***=AZ=MYSECOND.COM=FOO@MYFIRSTDOMAIN.com"`. Not sure if this is different issue than the original topic.

**Comment by Delevin888** [comment]
> Hi im using docker compose and how to add two domain names ? can someone guide me how to set up multiple domains?
> 
> ```
> version: '3.8'
> 
> services:
>   mailserver:
>     image: docker.io/mailserver/docker-mailserver:latest
>     container_name: mailserver
>     hostname: mail
>     # Change this to your domain, it is used for your email accounts (eg: user@example.com):
>     domainname: example.com
> ```

how about now? i have same question.

**Comment by aaronorosen2** [comment]
We are in need of a solution for this one too

**Comment by tinyaustralia** [comment]
> You are wasting maintainers' time:
> 
> https://github.com/docker-mailserver/docker-mailserver/blob/701037d3b2909db5b268ae4dbc171d70921600c3/README.md?plain=1#L181-L189
> 
> There is this file, in this repository's root directory, called `README`. No, capslock is not on, the file is **_literally_** called `README`, and guess what: you should read it. Oh my, the section is literally called "Support for Multiple Domains". Care to not use this issue tracker for things that are **_clearly well documented_**?

Why is he wasting maintainers time? He's simply asking a question. Don't be rude.

**Comment by polarathene** [comment]
**TL;DR:**
- You can add mail accounts with any domain you like and DMS will work with that provided the mail related DNS records for that domain point to the DMS FQDN. Do this via the `setup` command as per our [FAQ](https://docker-mailserver.github.io/docker-mailserver/v14.0/faq/#do-you-support-multiple-domains).
- Your mail domains have nothing to do with the DMS FQDN (_that you configure as `hostname` in `compose.yaml`_)
- Your DMS FQDN is distinct, it needs TLS certificate, your mail domains have no relevance to TLS interaction within DMS (_common misunderstanding_).
- If you really believe you need each mail domain to have vanity records instead of sharing a common FQDN for DMS, please [read this](https://github.com/docker-mailserver/docker-mailserver/pull/4155#issuecomment-2278861770). You'll create more problems for no good reason.

> Why is he wasting maintainers time? He's simply asking a question. Don't be rude.

Why are you responding to a comment from almost 3 years ago? That is rude. You notify anyone subscribed to this issue for something that has no relevance to the actual topic.

Please refrain from such behaviour. You have been called out for this serveral times now:
- https://github.com/docker-mailserver/docker-mailserver/issues/479#issuecomment-2249188398
- https://github.com/docker-mailserver/docker-mailserver/issues/1663#issuecomment-2299876551

I don't know what your motivations are to respond to old maintainer comments as you have been, but if this continues you will likely be issued a ban.

You are expected to use due diligence when seeking support, such as looking for this information on our docs or existing issues. Our issue report templates try to further emphasize this expectation with a checkbox and linking to a troubleshooting page on the docs. To not put in such basic efforts is disrespecting maintainers time.

---

As for the actual topic here itself... It's been covered as the comment you quoted states, we clearly document that the `setup` command will allow you to add any email address with your mail domains.

These do not have to be related to the `hostname` / FQDN for DMS as the mail server (_which each domain would reference for their mailserver related DNS records_).

A separate area of confusion that we sometimes see is a desire to have more than one `hostname` for the mail server, so that DNS records of each domain seems to not reference a foreign domain for the mail server. Usually this is from someone who is using DMS for multiple business clients as a service. This really does not make much sense and you will not see it supported by enterprise solutions, for more information see: https://github.com/docker-mailserver/docker-mailserver/pull/4155#issuecomment-2278861770

While the README contents has since moved into proper docs, we do [try to explain this on the docs](https://docker-mailserver.github.io/docker-mailserver/v14.0/usage) and in the [FAQ](https://docker-mailserver.github.io/docker-mailserver/v14.0/faq/#do-you-support-multiple-domains) (_this is where the [README entry was migrated to docs in Feb 2023](https://github.com/docker-mailserver/docker-mailserver/commit/853301338cf355d440561d1f17c45ebaf8d26301#diff-3288b20bd4420bc921919917b702f199137987db209c542ff20ac4500c1f9f28)_):

> ![image](https://github.com/user-attachments/assets/cdedee7f-c52b-4717-bfcf-3b84ff6c4ee0)

> ![image](https://github.com/user-attachments/assets/dccae605-8911-410e-92c8-87338456494a)

In the upcoming DMS v15 release, the docs will have the mult-domain concern [better clarified in an easy to find location](https://docker-mailserver.github.io/docker-mailserver/edge/config/account-management/overview/):

> ![image](https://github.com/user-attachments/assets/62047cd2-32af-40aa-a701-4a675c56b5a1)

**Comment by tinyaustralia** [comment]
> **TL;DR:**
> 
> * You can add mail accounts with any domain you like and DMS will work with that provided the mail related DNS records for that domain point to the DMS FQDN. Do this via the `setup` command as per our [FAQ](https://docker-mailserver.github.io/docker-mailserver/v14.0/faq/#do-you-support-multiple-domains).
> * Your mail domains have nothing to do with the DMS FQDN (_that you configure as `hostname` in `compose.yaml`_)
> * Your DMS FQDN is distinct, it needs TLS certificate, your mail domains have no relevance to TLS interaction within DMS (_common misunderstanding_).
> * If you really believe you need each mail domain to have vanity records instead of sharing a common FQDN for DMS, please [read this](https://github.com/docker-mailserver/docker-mailserver/pull/4155#issuecomment-2278861770). You'll create more problems for no good reason.
> 
> > Why is he wasting maintainers time? He's simply asking a question. Don't be rude.
> 
> Why are you responding to a comment from almost 3 years ago? That is rude. You notify anyone subscribed to this issue for something that has no relevance to the actual topic.
> 
> Please refrain from such behaviour. You have been called out for this serveral times now:
> 
> * [Can't login to mail server #479 (comment)](https://github.com/docker-mailserver/docker-mailserver/issues/479#issuecomment-2249188398)
> * [How to block all emails from specific domain or particular email address? #1663 (comment)](https://github.com/docker-mailserver/docker-mailserver/issues/1663#issuecomment-2299876551)
> 
> I don't know what your motivations are to respond to old maintainer comments as you have been, but if this continues you will likely be issued a ban.
> 
> You are expected to use due diligence when seeking support, such as looking for this information on our docs or existing issues. Our issue report templates try to further emphasize this expectation with a checkbox and linking to a troubleshooting page on the docs. To not put in such basic efforts is disrespecting maintainers time.
> 
> As for the actual topic here itself... It's been covered as the comment you quoted states, we clearly document that the `setup` command will allow you to add any email address with your mail domains.
> 
> These do not have to be related to the `hostname` / FQDN for DMS as the mail server (_which each domain would reference for their mailserver related DNS records_).
> 
> A separate area of confusion that we sometimes see is a desire to have more than one `hostname` for the mail server, so that DNS records of each domain seems to not reference a foreign domain for the mail server. Usually this is from someone who is using DMS for multiple business clients as a service. This really does not make much sense and you will not see it supported by enterprise solutions, for more information see: [#4155 (comment)](https://github.com/docker-mailserver/docker-mailserver/pull/4155#issuecomment-2278861770)
> 
> While the README contents has since moved into proper docs, we do [try to explain this on the docs](https://docker-mailserver.github.io/docker-mailserver/v14.0/usage) and in the [FAQ](https://docker-mailserver.github.io/docker-mailserver/v14.0/faq/#do-you-support-multiple-domains) (_this is where the [README entry was migrated to docs in Feb 2023](https://github.com/docker-mailserver/docker-mailserver/commit/853301338cf355d440561d1f17c45ebaf8d26301#diff-3288b20bd4420bc921919917b702f199137987db209c542ff20ac4500c1f9f28)_):
> 
> > ![image](https://private-user-images.githubusercontent.com/5098581/360728740-cdedee7f-c52b-4717-bfcf-3b84ff6c4ee0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzODAxNDYsIm5iZiI6MTcyNDM3OTg0NiwicGF0aCI6Ii81MDk4NTgxLzM2MDcyODc0MC1jZGVkZWU3Zi1jNTJiLTQ3MTctYmZjZi0zYjg0ZmY2YzRlZTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMDIyNDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTI2ZDE4MTJhNTM5YThiYzY2ODg2YzJlYmViZTFlMDAzYzUxNWUzNWU0MzhiM2ZhNDFjZjBlOGExNzYzOTk4MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kxrueB7QxIMUSyaeZs8sz3VIyeX1jX-6uT8iF5-yFNk)
> 
> > ![image](https://private-user-images.githubusercontent.com/5098581/360729039-dccae605-8911-410e-92c8-87338456494a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzODAxNDYsIm5iZiI6MTcyNDM3OTg0NiwicGF0aCI6Ii81MDk4NTgxLzM2MDcyOTAzOS1kY2NhZTYwNS04OTExLTQxMGUtOTJjOC04NzMzODQ1NjQ5NGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMDIyNDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDEwYzFjZjIyNmYyNGI4N2Q2Nzk0Njk5Njc2MTZiYmE0ZmEyZDRkZmE5YjI5ZDljMzcyZGEwMjU0Y2ZjZTVmNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.TrlRM-4OP-47MZMIDPlLh6ES26NvVd8ePM6I6u8tr2U)
> 
> In the upcoming DMS v15 release, the docs will have the mult-domain concern [better clarified in an easy to find location](https://docker-mailserver.github.io/docker-mailserver/edge/config/account-management/overview/):
> 
> > ![image](https://private-user-images.githubusercontent.com/5098581/360728553-62047cd2-32af-40aa-a701-4a675c56b5a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzODAxNDYsIm5iZiI6MTcyNDM3OTg0NiwicGF0aCI6Ii81MDk4NTgxLzM2MDcyODU1My02MjA0N2NkMi0zMmFmLTQwYWEtYTcwMS00YTY3NWM1NmI1YTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMDIyNDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzNlMDQ3YTAwOWNhMGFiMDYzNjgxOWRiMDM3OGNmY2EyNjY2MTgwNDdlMTQ5ZDFiZDIwNzNlMDc0MzVlMWYwNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.c6OPSex0u7NL_FYgNk5DNMBRfHR9GWSImdq4XHjjJKI)

There are people out there that are not developers and are familiar with Github. Just because I comment on a thread that is a couple years old, and my comment in fact is thoughtful to the OP asking the question, why do you have to threaten? Just chill man.

**Comment by polarathene** [comment]
> There are people out there that are not developers and not familiar with Github.

You're familiar enough to be deploying a mailserver with Docker. That's more technical skill than most have.

The whole purpose of this project is to make this easier for users like yourself, maintainers volunteer considerable time into this project. I've especially done so for the docs and assisting users in issues and discussions.

> Just because I comment on a thread that is a couple years old, and my comment in fact is thoughtful to the OP asking the question, why do you have to threaten?

It is not a threat. You may not be familiar with how your comments are bad etiquette on this platform, hence I have tried to make you aware of this already in the past when you have done so. I do not resort to bans if they can be avoided, but I can't speak for other maintainers if you continue to engage like you are.




> And please enlighten me as to how commenting on a 3 year old thread is rude? Ban me if you want bro. Bigger things in life than dealing with people on a power trip.

This is not a chat platform. Your comments will notify all users subscribed to an issue (_not just the ones that have posted comments_). It is unhelpful to comment on issues that are years old as quite a lot can change over that time.

When you do comment you often have nothing of value to contribute, with this issue and the previous one your comments did not express any relevance to the actual issue even affecting you.

Nobody is on a power trip. However if all you're going to do is be disruptive and disrespectful to the community here after repeated attempts to discourage how you're choosing to engage with the project... well what else are we to do?

> Finally, you yourself are being hypocritical since you have also called me rude and commented on a 3 year old post..

I responded because I was notified each time. My responses to you have typically been cordial and contributed information related to the actual issue.

I could ignore you, but I rather try assist users when possible. What I do not appreciate is how your comments tend to be  poor etiquette and I've tried to reach out to you to discourage that.

I understand that you may not have realized why your own actions are considered rude here, but I hope that you'll choose to respect what is being asked of you, now that you know better.

---

I'm not interested in arguing or debating this further.
- Please respect the community here, or cease commenting solely to criticize (_it helps nobody here with the actual issue_).
- If you have something to contribute to an issue that is beneficial to the actual problem, that is often welcome.
- If you are experiencing a problem on an issue that is closed and not had any recent activity, please open a new bug report or discussion and link to it instead as a reference.

Cheers

**Comment by tinyaustralia** [comment]
> > There are people out there that are not developers and not familiar with Github.
> 
> You're familiar enough to be deploying a mailserver with Docker. That's more technical skill than most have.
> 
> The whole purpose of this project is to make this easier for users like yourself, maintainers volunteer considerable time into this project. I've especially done so for the docs and assisting users in issues and discussions.
> 
> > Just because I comment on a thread that is a couple years old, and my comment in fact is thoughtful to the OP asking the question, why do you have to threaten?
> 
> It is not a threat. You may not be familiar with how your comments are bad etiquette on this platform, hence I have tried to make you aware of this already in the past when you have done so. I do not resort to bans if they can be avoided, but I can't speak for other maintainers if you continue to engage like you are.
> 
> > And please enlighten me as to how commenting on a 3 year old thread is rude? Ban me if you want bro. Bigger things in life than dealing with people on a power trip.
> 
> This is not a chat platform. Your comments will notify all users subscribed to an issue (_not just the ones that have posted comments_). It is unhelpful to comment on issues that are years old as quite a lot can change over that time.
> 
> When you do comment you often have nothing of value to contribute, with this issue and the previous one your comments did not express any relevance to the actual issue even affecting you.
> 
> Nobody is on a power trip. However if all you're going to do is be disruptive and disrespectful to the community here after repeated attempts to discourage how you're choosing to engage with the project... well what else are we to do?
> 
> > Finally, you yourself are being hypocritical since you have also called me rude and commented on a 3 year old post..
> 
> I responded because I was notified each time. My responses to you have typically been cordial and contributed information related to the actual issue.
> 
> I could ignore you, but I rather try assist users when possible. What I do not appreciate is how your comments tend to be poor etiquette and I've tried to reach out to you to discourage that.
> 
> I understand that you may not have realized why your own actions are considered rude here, but I hope that you'll choose to respect what is being asked of you, now that you know better.
> 
> I'm not interested in arguing or debating this further.
> 
> * Please respect the community here, or cease commenting solely to criticize (_it helps nobody here with the actual issue_).
> * If you have something to contribute to an issue that is beneficial to the actual problem, that is often welcome.
> * If you are experiencing a problem on an issue that is closed and not had any recent activity, please open a new bug report or discussion and link to it instead as a reference.
> 
> Cheers

Buddy, look. This is just your opinion and frankly you're trying to maintain your position here, which is fine. I did not disrespect anyone and you are being a hypocrit. You are engaging in a debate here on a 3 year old post trying to pep talk me on etiquette. You have engaged in pettyness for a comment I left about a poster who was rude to the OP. If you don't like it, then ban me. Or just turn off commenting on threads that are older than X amount of time. I'm not wasting more time on this.
